docs/installation/autoware/docker-installation.md

```diff
@@ -16,21 +16,137 @@
+
+# Restart the Docker daemon to complete the installation after setting the default runtime:
+sudo systemctl restart docker
+
+# At this point, a working setup can be tested by running a base CUDA container:
+sudo docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi
+
+# This should result in a console output shown below:
+# +-----------------------------------------------------------------------------+
+# | NVIDIA-SMI 450.51.06    Driver Version: 450.51.06    CUDA Version: 11.0     |
+# |-------------------------------+----------------------+----------------------+
+# | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
+# | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
+# |                               |                      |               MIG M. |
+# |===============================+======================+======================|
+# |   0  Tesla T4            On   | 00000000:00:1E.0 Off |                    0 |
+# | N/A   34C    P8     9W /  70W |      0MiB / 15109MiB |      0%      Default |
+# |                               |                      |                  N/A |
+# +-------------------------------+----------------------+----------------------+
+#
+# +-----------------------------------------------------------------------------+
+# | Processes:                                                                  |
+# |  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
+# |        ID   ID                                                   Usage      |
+# |=============================================================================|
+# |  No running processes found                                                 |
+# +-----------------------------------------------------------------------------+
+```
+
+Install Docker Compose:
+
+```bash
+# Taken from: https://docs.docker.com/compose/install/#install-compose-on-linux-systems
+
+# Run this command to download the current stable release of Docker Compose:
+sudo curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+
+# Apply executable permissions to the binary:
+sudo chmod +x /usr/local/bin/docker-compose
+
+# Optionally Install command completion
+# Place the completion script in /etc/bash_completion.d/
+sudo curl \
+    -L https://raw.githubusercontent.com/docker/compose/1.29.2/contrib/completion/bash/docker-compose \
+    -o /etc/bash_completion.d/docker-compose
+
+# Reload your terminal. You can close and then open a new terminal, or reload your setting with source ~/.bashrc command in your current terminal.
+source ~/.bashrc
 
-    Be very careful with this method. Make sure you read and confirmed all the steps in the Ansible configuration before using it.
+# Test the installation.
+docker-compose --version
+```
+
+Install rocker:
+
+```bash
+# Taken from: https://github.com/osrf/rocker#installation
```

Comment:
Should we add a step about the ROS APT repository for consistency?

```bash
sudo apt update && sudo apt install curl gnupg lsb-release
sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
```

```bash
echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(source /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
```

The other dependencies have their APT repositories listed.

Reply:
@esteve why add ROS APT repository for docker installation? Isn't it already inside the docker?

Reply:
Added in: https://github.com/autowarefoundation/autoware/pull/141/files#diff-41f1e40341b8d84975bb0de78a9fd14351611575fc472827065247bd41116246R131-R139